common/types.ts

```diff
@@ -0,0 +1,56 @@
+// This file contains common types used through out the project.
+import { Book, Post, Course, Dept } from "@prisma/client";
+
+export type CourseWithDept = Course & {
+  dept: Dept;
+};
+
+export type BookWithPostWithUserWithCourseWithDept = Book & {
+  courses: CourseWithDept[];
+  posts: Post[];
+};
+
+export interface GoogleBook {
```

Comment:
Consider changing this to `export type GoogleBook = books_v1.Schema$Volume['volumeInfo']` (and similar for the other two Google books types) so we don't have maintain these types ourselves

Reply:
Done in latest commit.